src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -600,76 +499,40 @@ void MergeTreeData::setTTLExpressions(const ColumnsDescription & new_columns,
             for (const auto & col : getColumnsRequiredForSortingKey())
                 columns_ttl_forbidden.insert(col);
 
-        for (const auto & [name, ast] : new_column_ttls_asts)
+        for (const auto & [name, ttl_description] : new_column_ttls)
         {
             if (columns_ttl_forbidden.count(name))
                 throw Exception("Trying to set TTL for key column " + name, ErrorCodes::ILLEGAL_COLUMN);
-            else
-            {
-                auto new_ttl_entry = TTLDescription::getTTLFromAST(ast, new_columns, global_context, getPrimaryKey());
```

Comment:
`getTTLFromAST` contains a lot of checks. Maybe we still should call it if we want to slit check and set stages?

Reply:
It's already called in AlterCommands and registerStorageMergeTree functions, we will get the error earlier. The main idea was to get rid of parsing structures inside storage. Here we check only "storage-specific" errors with already parsed and pre-checked TTL structure.